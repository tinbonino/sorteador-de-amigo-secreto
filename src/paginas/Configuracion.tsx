import Formulario from "../components/Formulario";
import Footer from "../components/Footer";
import ListaParticipantes from "../components/ListaParticipantes";
import Card from "../components/Card";


const Configuracion = () => {
    return (
        <Card>  
            <section>
                <h2>Vamos a Comenzar!</h2>
                <Formulario/>
                <ListaParticipantes/>
                <Footer/>
            </section>
        </Card>
    )
  
}

export default Configuracion