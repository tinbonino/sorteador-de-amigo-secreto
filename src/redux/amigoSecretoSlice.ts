import { createSlice, PayloadAction } from "@reduxjs/toolkit";

interface AmigoSecreto {
    names: string[]
}

const initialState: AmigoSecreto = {
    names: []
};

const amigoSecretoSlice = createSlice({
    name: "amigoSecreto",
    initialState,
    reducers: {
        addName: (state, action: PayloadAction<string>) =>{
            if (action.payload && !state.names.includes(action.payload)) {
                state.names.push(action.payload)
            } 
        },
        resetNames: (state) =>{
            state.names = []
        }
    }
});

export const {addName,resetNames} = amigoSecretoSlice.actions;
export default amigoSecretoSlice.reducer;