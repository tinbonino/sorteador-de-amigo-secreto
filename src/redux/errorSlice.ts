import {createSlice} from "@reduxjs/toolkit";

interface ErrorState {
    message: string | null;
}

const initialState: ErrorState = {
    message: null
};

const errorSlice = createSlice({
    name: "error",
    initialState,
    reducers: {
        setError: (state, action) => {
            state.message = action.payload;
        },
        clearError: (state) => {
            state.message=null;
        }
    }
});

export const {setError,clearError} = errorSlice.actions;
export default errorSlice.reducer;

