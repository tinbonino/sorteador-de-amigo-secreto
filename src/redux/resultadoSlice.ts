import { createSlice, PayloadAction } from "@reduxjs/toolkit";

interface ResultadoState {
    sorteo: Map<string,string>;
}

const initialState: ResultadoState = {
    sorteo: new Map(),
}

const resultadoSlice = createSlice(
    {
        name:"resultadoAmigoSecreto",
        initialState,
        reducers: {
            definirResultado: (state,action: PayloadAction<Map<string,string>>) =>{
                state.sorteo=action.payload;
            },
            limpiarResultado: (state)=> {
                state.sorteo=new Map();
            }
        }

});

export const {definirResultado, limpiarResultado} = resultadoSlice.actions;
export default resultadoSlice.reducer;