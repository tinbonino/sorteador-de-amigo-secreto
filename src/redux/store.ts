import { configureStore } from "@reduxjs/toolkit";
import amigoSecretoReducer from "./amigoSecretoSlice";
import errorReducer from "./errorSlice";



export const store = configureStore({
    reducer: {
        amigoSecreto:amigoSecretoReducer,
        error: errorReducer
    }
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;