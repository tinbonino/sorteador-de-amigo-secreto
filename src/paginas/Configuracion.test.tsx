import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { store } from "../redux/store";
import Configuracion from "./Configuracion";


import { vi } from "vitest";

const mockNavigate = vi.fn();

vi.mock("react-router-dom", () => ({
  useNavigate: () =>mockNavigate,
})
);



describe("La página de configuración", () => {
  test("Debe ser renderizada correctamente", () => {
    const {container} = render(
        <Provider store={store}>
            <Configuracion/>
        </Provider>
    )

    expect(container).toMatchSnapshot()

  })
})