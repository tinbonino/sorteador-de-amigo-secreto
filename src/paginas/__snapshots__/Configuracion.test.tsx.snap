// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`La página de configuración > Debe ser renderizada correctamente 1`] = `
<div>
  <div
    class="card"
  >
    <section>
      <h2>
        Vamos a Comenzar!
      </h2>
      <form>
        <div
          class="grupo-input-btn"
        >
          <input
            placeholder="Ingrese los nombres de los participantes"
            type="text"
            value=""
          />
          <button
            disabled=""
          >
            Agregar
          </button>
        </div>
      </form>
      <ul />
      <footer
        class="footer-configuraciones"
      >
        <button
          class="boton"
          disabled=""
        >
          Iniciar Juego
        </button>
        <img
          alt="Bolsa de compras"
          src="/imagenes/bolsas.png"
        />
      </footer>
    </section>
  </div>
</div>
`;
